superset/connectors/sqla/models.py

```diff
@@ -484,30 +487,34 @@ def get_from_clause(self, template_processor=None, db_engine_spec=None):
             return TextAsFrom(sa.text(from_sql), []).alias('expr_qry')
         return self.get_sqla_table()
 
-    def adhoc_metric_to_sa(self, metric, cols):
+    def adhoc_metric_to_sa(self, metric, cols, db_engine_spec):
```

Comment:
I think `db_engine_spec` is also in scope here through `self.table.database.db_engine_spec`